lib/Providers/Qr/BaconQrCodeProvider.php

```diff
@@ -1,51 +1,43 @@
 <?php
 
+declare(strict_types=1);
+
 namespace RobThree\Auth\Providers\Qr;
 
-use BaconQrCode\Writer;
-use BaconQrCode\Renderer\ImageRenderer;
-use BaconQrCode\Renderer\RendererStyle\RendererStyle;
-use BaconQrCode\Renderer\RendererStyle\Fill;
 use BaconQrCode\Renderer\Color\Rgb;
-use BaconQrCode\Renderer\RendererStyle\EyeFill;
-
 use BaconQrCode\Renderer\Image\EpsImageBackEnd;
 use BaconQrCode\Renderer\Image\ImageBackEndInterface;
 use BaconQrCode\Renderer\Image\ImagickImageBackEnd;
 use BaconQrCode\Renderer\Image\SvgImageBackEnd;
+use BaconQrCode\Renderer\ImageRenderer;
+
+use BaconQrCode\Renderer\RendererStyle\EyeFill;
+use BaconQrCode\Renderer\RendererStyle\Fill;
+use BaconQrCode\Renderer\RendererStyle\RendererStyle;
+use BaconQrCode\Writer;
+use RuntimeException;
 
 class BaconQrCodeProvider implements IQRCodeProvider
 {
-    private $borderWidth = 4; // default from Bacon QR Code
-    private $backgroundColour;
-    private $foregroundColour;
-    private $format;
-
     /**
      * Ensure we using the latest Bacon QR Code and specify default options
-     *
-     * @param int $borderWidth space around the QR code, 4 is the default from Bacon QR Code
-     * @param string $backgroundColour hex reference for the background colour
-     * @param string $foregroundColour hex reference for the foreground colour
-     * @param string $format the desired output, png or svg
      */
-    public function __construct($borderWidth = 4, $backgroundColour = '#ffffff', $foregroundColour = '#000000', $format = 'png')
-    {
-        if (! class_exists(ImagickImageBackEnd::class)) {
-            throw new \RuntimeException('Make sure you are using version 2 of Bacon QR Code');
+    public function __construct(
+        private int $borderWidth = 4,
+        private string|array $backgroundColour = '#ffffff',
+        private string|array $foregroundColour = '#000000',
+        private string $format = 'png',
+    ) {
+        if (!class_exists(ImagickImageBackEnd::class)) {
+            throw new RuntimeException('Make sure you are using version 2 of Bacon QR Code');
```

Comment:
This check could be removed as it's not possible to install `bacon/bacon-qr-code@^1` on PHP >= 8